sdv/constraints/tabular.py

```diff
@@ -300,24 +325,30 @@ def _fit(self, table_data):
             table_data (pandas.DataFrame):
                 The Table data.
         """
-        if self._high_is_scalar is None:
-            self._high_is_scalar = self._high not in table_data.columns
-        if self._low_is_scalar is None:
-            self._low_is_scalar = self._low not in table_data.columns
-
-        if self._high_is_scalar and self._low_is_scalar:
-            raise TypeError('`low` and `high` cannot be both scalars at the same time')
-        elif self._low_is_scalar:
-            self.constraint_columns = (self._high,)
-            self._dtype = table_data[self._high].dtype
-        elif self._high_is_scalar:
-            self.constraint_columns = (self._low,)
-            self._dtype = table_data[self._low].dtype
+        if self._scalar == 'high':
+            exists = [column in table_data.columns for column in self._low]
+            if not all(exists):
+                raise KeyError(f'The `low` columns {list(compress(self._low, exists))} '
+                               'were not found in table_data.')
+        elif self._scalar == 'low':
+            exists = [column in table_data.columns for column in self._high]
+            if not all(exists):
+                raise KeyError(f'The `high` columns {list(compress(self._high, exists))} '
+                               'were not found in table_data.')
         else:
-            self._dtype = table_data[self._high].dtype
-
-        self._column_to_reconstruct = self._get_column_to_reconstruct()
-        self._diff_column = self._get_diff_column_name(table_data)
+            for column in self._low:
```

Comment:
This block and the one below could be moved to a method that is called twice, passing `_low` and `_high` (and maybe the argument name to add to the error message)